test/Microsoft.AspNetCore.SignalR.Tests/WebSocketsTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    [CollectionDefinition(Name)]
+    public class FunctionalTestsCollection : ICollectionFixture<ServerFixture>
+    {
+        public const string Name = "FunctionalTests";
+    }
+
+    [Collection(FunctionalTestsCollection.Name)]
+    public class WebSocketsTests
+    {
+        private readonly ServerFixture _serverFixture;
+
+        public WebSocketsTests(ServerFixture serverFixture)
+        {
+            _serverFixture = serverFixture;
+        }
+
+        [Fact]
+        public async Task CheckEchoWebSockets()
+        {
+            const string message = "Hello, World!";
+
+            using (var ws = new ClientWebSocket())
+            {
+                var url = _serverFixture.BaseUrl.Replace("http", "ws") + "echo/ws";
+
+                await ws.ConnectAsync(new Uri(url), CancellationToken.None);
+                var bytes = Encoding.UTF8.GetBytes(message);
+                await ws.SendAsync(new ArraySegment<byte>(bytes), WebSocketMessageType.Binary, true, CancellationToken.None);
```

Comment:
What is the scenario for this? What would it prevent? If the server goes down the websocket will be closed..